### PR TITLE
[15.0][MIG] account_payment_promissory_note: Migration to 15.0

### PR DESCRIPTION
**Pull request by cesar-tecnativa**
@Tecnativa TT36428

### CODE REVIEW

**Comment by LudLaf** [comment]
@cesar-tecnativa, pre-commit was failing due to a redundant attribute string. This has been fixed.

**Comment by sergio-teruel** [comment]
@cesar-tecnativa Plase check tests

**Comment by CarlosRoca13** [comment]
I have finished the migration of this module please @sergio-teruel @carlosdauden review this

File: account_payment_promissory_note/wizard/account_register_payments.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    _name = "account.payment.register"
+    _inherit = ["account.payment.register", "account.promissory.note.mixin"]
+
+    def _create_payments(self):
+        payments = super()._create_payments()
+        for payment in payments:
+            if not self.date_due:
+                invoices = payment.reconciled_invoice_ids
+                max_date = max(invoices.mapped("invoice_date_due"))
+                payment.write(
```
Comment (carlosdauden) [inline account_payment_promissory_note/wizard/account_register_payments.py]: Use the update method or field assign directly to take advantage of improvements in the framework
```suggestion
                payment.update(
```

File: account_payment_promissory_note/wizard/account_register_payments.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                    {"promissory_note": self.promissory_note, "date_due": max_date}
+                )
+            else:
+                payment.write(
```
Comment (carlosdauden) [inline account_payment_promissory_note/wizard/account_register_payments.py]: Use the update method or field assign directly to take advantage of improvements in the framework
```suggestion
                payment.update(
```

**Review by carlosdauden — CHANGES_REQUESTED** [review]

**Comment by CarlosRoca13** [comment]
@carlosdauden changed!

**Review by sergio-teruel — APPROVED** [review]

**Review by carlosdauden — APPROVED** [review]
Unrelated pre-commit error

**Comment by pedrobaeza** [comment]
Please update copier template in another PR for solving the pre-commit error.

**Comment by CarlosRoca13** [comment]
Done @pedrobaeza

**Comment by pedrobaeza** [comment]
@CarlosRoca13 I think you haven't updated the copier template in the repository.

**Pull request merged**